spec/system/admin/reports/orders_and_fulfillment_spec.rb

```diff
@@ -438,5 +440,167 @@
         end
       end
     end
+
+    describe "Saving report rendering options" do
+      let(:report_title) do
+        "Order Cycle Supplier Totals"
+      end
+
+      let(:second_report_title) do
+        "Order Cycle Supplier Totals by Distributor"
+      end
+
+      let(:display_header_row_selector) do
+        "#display_header_row"
+      end
+
+      let(:display_summary_row_selector) do
+        "#display_summary_row"
+      end
+
+      let(:columns_dropdown_selector) do
+        'div[data-multiple-checked-select-target="button"]'
+      end
+      let(:producer_column_checkbox_selector) do
+        "input[type='checkbox'][name='fields_to_show[]'][value='producer']"
+      end
+
+      let(:product_column_checkbox_selector) do
+        "input[type='checkbox'][name='fields_to_show[]'][value='product']"
+      end
+
+      context "Switching between reports" do
+        context "Display options" do
+          it "should store display options for every report separately" do
+            # Step 1: Update report rendering options on two reports
+            click_link report_title
+            display_header_row = find(display_header_row_selector)
+            display_summary_row = find(display_summary_row_selector)
+            expect(display_header_row.checked?).to eq(false)
+            expect(display_summary_row.checked?).to eq(true)
+            display_header_row.set(true)
```

Comment:
Just FYI in general, I think it's better to test with user interactions. For example I think we can do something like this with Capybara:

```suggestion
            expect(page).to have_checked_field('SUMMARY ROW"') 
            check "SUMMARY ROW"
```

This means we're not tied to the actual implementation with a specific html ID. Also it can be less code, and easier to understand the code.

But I'm not all that familiar with our testing suite so am not sure if this change is necessary right now.

Reply:
I copied this from  [existing tests].(https://github.com/openfoodfoundation/openfoodnetwork/blob/master/spec/system/admin/reports/orders_and_fulfillment_spec.rb#L127)

I like your idea, this will make tests less vulnerable to changes in other parts of the page.

Reply:
Done